src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDurationMapping.java

```diff
@@ -19,54 +19,93 @@
         OverrideFields overrideFields = OverridesMapping.getOverrideFields(caseData);
 
-        if (nonNull(overrideFields.getDuration()) && overrideFields.getDuration().intValue() >= MIN_HEARING_DURATION) {
+        if (nonNull(overrideFields.getDuration())
+            && overrideFields.getDuration().intValue() >= MIN_HEARING_DURATION
+        ) {
             return overrideFields.getDuration().intValue();
         }
 
-        Integer duration = getHearingDurationAdjournment(caseData);
+        Integer duration = getHearingDurationAdjournment(caseData, referenceDataServiceHolder);
         if (isNull(duration)) {
             duration = getHearingDurationBenefitIssueCodes(caseData, referenceDataServiceHolder);
         }
 
         return nonNull(duration) ? duration : DURATION_DEFAULT;
     }
 
-    public static Integer getHearingDurationAdjournment(SscsCaseData caseData) {
+    public static Integer getHearingDurationAdjournment(
+        SscsCaseData caseData,
+        ReferenceDataServiceHolder referenceDataServiceHolder
+    ) {
+        return referenceDataServiceHolder.isAdjournmentFlagEnabled() // TODO remove when flag enabled
+            ? getHearingDurationAdjournmentNew(caseData)
+            : getHearingDurationAdjournmentOld(caseData);
+    }
 
-        if (DURATION_TYPE_NON_STANDARD_TIME_SLOT.equalsIgnoreCase(caseData.getAdjournCaseNextHearingListingDurationType())
-            && isNotBlank(caseData.getAdjournCaseNextHearingListingDuration())) {
-            int duration = Integer.parseInt(caseData.getAdjournCaseNextHearingListingDuration());
+    private static Integer getHearingDurationAdjournmentOld(SscsCaseData caseData) {
+        String adjournCaseNextHearingListingDuration = caseData.getAdjournCaseNextHearingListingDuration();
 
-            if (DURATION_UNITS_SESSIONS.equalsIgnoreCase(caseData.getAdjournCaseNextHearingListingDurationUnits())
-                && duration >= MIN_HEARING_SESSION_DURATION) {
-                return duration * DURATION_SESSIONS_MULTIPLIER;
-            } else if (DURATION_UNITS_MINUTES.equalsIgnoreCase(caseData.getAdjournCaseNextHearingListingDurationUnits())
-                && duration >= MIN_HEARING_DURATION) {
+        if (isNotBlank(adjournCaseNextHearingListingDuration)
+            && Integer.parseInt(adjournCaseNextHearingListingDuration) >= MIN_HEARING_DURATION
+        ) {
+            if (DURATION_UNITS_SESSIONS.equalsIgnoreCase(caseData.getAdjournCaseNextHearingListingDurationUnits())) {
+                return Integer.parseInt(adjournCaseNextHearingListingDuration) * DURATION_SESSIONS_MULTIPLIER;
+            }
+            if (DURATION_UNIT_HOURS.equalsIgnoreCase(caseData.getAdjournCaseNextHearingListingDurationUnits())) {
+                // TODO Adjournments - check no other measurement than hours, sessions and null
+                return Integer.parseInt(adjournCaseNextHearingListingDuration) * DURATION_HOURS_MULTIPLIER;
+            }
+        }
+
+        return null;
```

Comment:
No change, just wondering if returning null has any undesired impact here? I note we're already returning null in other methods.

Also, any reason to remove if/else if for multiple if statements? I'm not fussed on which is being used when returning within them, as you're not using additional overhead for the other ifs when a condition is met, so all good, just minor comment really.